packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -468,6 +469,19 @@ export function injectIntoGlobalHook(globalObject: any): void {
       return id;
     };
 
+    // Do the same for any already injected roots.
+    // This is useful if ReactDOM has already been initialized.
+    // https://github.com/facebook/react/issues/17626
+    hook.renderers.forEach((injected, id) => {
+      if (
+        typeof injected.scheduleRefresh === 'function' &&
+        typeof injected.setRefreshHandler === 'function'
+      ) {
```

Comment:
I've applied these changes locally in my application (I had a pretty identical local branch going 😅), and it doesn't seem to work. The renderer instance has none of these properties at this point (react @ 16.8.6):

<img width="857" alt="Screen Shot 2019-12-16 at 5 36 32 PM" src="https://user-images.githubusercontent.com/3022244/70957343-b2480100-202a-11ea-93cd-904bf6c02958.png">

Reply:
Can you make a minimal reproducing project?

Reply:
@mxmul Oh wait, this is expected — these were only added in 16.10 or so. You need to try with the latest version.